src/context/propagation/encoder.rs

```diff
@@ -23,11 +23,11 @@ pub fn encode_propagation(context: &TracingContext, endpoint: &str, address: &st
     let mut res = String::new();
 
     res += "1-";
-    res += format!("{}-", encode(context.trace_id.to_string())).as_str();
-    res += format!("{}-", encode(context.trace_segment_id.to_string())).as_str();
+    res += format!("{}-", encode(context.trace_id())).as_str();
+    res += format!("{}-", encode(context.trace_segment_id())).as_str();
     res += format!("{}-", context.peek_active_span_id().unwrap_or(0)).as_str();
-    res += format!("{}-", encode(context.service.as_str())).as_str();
-    res += format!("{}-", encode(context.service_instance.as_str())).as_str();
+    res += format!("{}-", encode(context.service())).as_str();
+    res += format!("{}-", encode(context.service_instance())).as_str();
     res += format!("{}-", encode(endpoint)).as_str();
     res += &encode(address);
```

Comment:
Curiously, is multi `+` to link string and format multiple times good in performance? Or should we format once to use one format to link all parts?